orchestra/static/orchestra/timing/timer/directives.js

```diff
@@ -0,0 +1,105 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('orchestra.timing')
+    .directive('workTimer', function() {
+      return {
+        restrict: 'E',
+        scope: {
+          taskId: '=taskId',
+        },
+        bindToController: true,
+        controllerAs: 'workTimer',
+        templateUrl: '/static/orchestra/timing/timer/timer.html',
+        controller: function($http, $location, $scope, $timeout, timeEntries) {
+          var workTimer = this;
+
+          workTimer.popoverTemplate = '/static/orchestra/timing/timer/popover.html';
+
+          var resetTimer = function() {
+            workTimer.timerRunning = false;
+            workTimer.description = undefined;
+            workTimer.timeElapsed = moment.duration();
+          };
+
+          workTimer.toggleTimer = function() {
+            if (workTimer.toggledOn) {
```

Comment:
To decouple the display (the toggle button to turn the timer on and off) from the state of the timer—we don't want to immediately say the timer isn't running anymore before we find out from the server if that's actually true.